lore/stores/base.py

```diff
@@ -47,10 +49,17 @@ def key(self, *args, **kwargs):
         stack = inspect.stack()
         caller = kwargs.pop('caller', stack[-2])
         instance = kwargs.pop('instance', self)
+        if  len(args) > 0 and isinstance(args[0], sqlalchemy.sql.Selectable):
```

Comment:
Is it possible to have a more generic solution rather than special casing for Selectable? Was the issue that selectable doesn't have a stable string representation?